### DISCUSSION TITLE
object with vector3

### DISCUSSION THREAD

**Discussion by TomerAnzu** [gwaredd/unium#129] [Q&A] [answered]
I have some troubles to transfer object with object property(vector3) in JS to unity.

I want to do something like that:

JS:

let object = JSON.stringify(
      { 
        name: 'test',
        position: {x:2, y:1, z:1},
      }
 );
 
 await ...(...Game.ScriptName.Create($(object)));
 
 c#:
 
 I have TestObject class:
 
[Serializable]
public class TestObject
{
    public string name;
    public Vector3 position;
}

and in the ScriptName:

public void Create(TestObject testObject){
    print(testObject.position)
}

how can I do that?

**Comment by gwaredd** [comment]
Hey, Hmm - looks like there is a bug. I will take a look

**Comment by gwaredd** [comment, marked as answer]
Fixed by https://github.com/gwaredd/unium/commit/3fd5dd2f74618a82c7ca9636e3c908463e8ef341

**Reply by TomerAnzu** [reply]
thanks for the quick response and support!

i will give it  a try to see how this one works.

just a quick question, can you send this function complex object and parameters as a rest  POST request? or this one is a GET only as it used to be?

i'm asking because i know that there are some limitations of the HTTP GET length message, and there are times when we send long json object which may exceeded this limitation.

**Comment by gwaredd** [comment]
No, GET is the only method supported. Originally the use case was to be able to peek and poke directly from a browser so anything more sophisticated wasn't required and I liked the simplicity.

However I can see the limitation and it should be relatively straightforward to add.

I would keep the same path format but allow named parameters, something like.

`/node/obj.com.func(@a,@b)`

Pass the request body as an extra parameter when [creating the query](https://github.com/gwaredd/unium/blob/02e2610b9b02141b7ece8861038d1964271e2cc8/Assets/Unium/Routing/HandlerGQL.cs#L63).

Then when it's [resolving the function arguments](https://github.com/gwaredd/unium/blob/02e2610b9b02141b7ece8861038d1964271e2cc8/Assets/Unium/GQL/Query/ActionInvoke.cs#L104), you just need to add some extra logic to grab the contents of the body when you see a `@`.

This isn't something I have time to do but I am happy to review a pull request.

**Reply by gwaredd** [reply]
Alternatively ... you can hack the size of the HTTP header [here](https://github.com/gwaredd/unium/blob/02e2610b9b02141b7ece8861038d1964271e2cc8/Assets/Unium/Core/gw.proto.http/HttpRequest.cs#L20) ;)

IIRC here is no limitations in the protocol, it's more of a practical implementation restriction for webservers - 8k or 16k are typical.
